smps.proto

```diff
@@ -0,0 +1,177 @@
+        message Latency {
+            int32 min = 1;
+            int32 average = 2;
+            int32 p50 = 3;
+            int32 p90 = 4;
+            int32 p99 = 5;
+            int32 max = 6;    
+        }  
+        
+        bool internal = 1;
+        Protocol protocol = 2;
+        int32 connections = 3;
+        int64 rps = 4;
+        Latency latencies_ms = 5;
+    }
+    
+    message Metric {
+        
+        message IngressGateway {
+            int32 count = 1;
+            // cpu in mCores
+            int32 cpu_mCores = 2;
+            // memory in MB
+            int32 mem_mb = 3;
+            // traffic sent thru ingress gateway
+            int32 rps = 4;
+            // Total bytes sent thru ingress
+            int32 bps = 5;
+        }
+        
+        message Sidecar {
+            int32 count = 1;
+            // cpu used by all sidecars except ingress / egress
+            int32 cpu_mCores = 2;
+            int32 mem_mb = 3;
+            // Total rps traversing all sidecars
+            int32 rps = 4;
+            // Total bytes sent thru sidecars
+            int32 bps = 5;
+        }
+        
+        message MeshTelemetry {
+            int32 count = 1;
+            // cpu used by all mesh_telemetry pods
+            int32 cpu_mCores = 2;
+            int32 mem_mb = 3;
+            // Total rps traversing all proxies (sidecars+ingress+egress)
+            int32 rps = 4;
+        }
+        
+        message MeshPolicy {
+            int32 count = 1;
+            // cpu used by all mesh_policy pods
+            int32 cpu_mCores = 2;
+            int32 mem_mb = 3;
+            // Total rps traversing all proxies (sidecars+ingress+egress)
+            int32 rps = 4;
+            int32 cache_hit_rate = 5;
+        }
+        
+        message MeshControlPlane {
+            int32 count = 1;
+            // cpu used by mesh_pilot pods
+            int32 cpu_mCores = 2;
+            int32 mem_mb = 3;
+            int32 endpoints = 4;
+            // services + service entries
+            int32 services = 5;  
+            int32 sidecars = 6;
+            int32 virtual_services = 7;
+            int32 destination_rules = 8;
+            // how long does it take a listener change to propagate to 90% proxies
+            int32 lds_latency_ms = 9;
+            // how long does it take for cluster change to propagate to 90% proxies
+            int32 cds_latency_ms = 10;
+        }
+        
+        IngressGateway ingress_gateway = 1;
+        Sidecar sidecar = 2;
+        MeshTelemetry mesh_telemetry = 3;
+        MeshPolicy mesh_policy = 4;
+        MeshControlPlane mesh_control_plane = 5;        
+    }
+
+    // additional individual workloads should be listed here
+    // Only sidecar metrics are measured
+    message IndividualWorkload {
+        string name = 1;
+        int32 count = 2;
+        int32 cpu_mCores = 3;
+        int32 mem_mb = 4;
+    }
+    
+    // Spec version
+    string smps_version = 1;
```

Comment:
Gents, what say you to starting the spec version at v0.1.0? Or something much less that v1.0?

Reply:
Hmm.. true.. We have a lot of things to finalize

Reply:
Also, something we want to clarify would be that which of these elements are lists?

Reply:
I second that @leecalcote !!! versioning should start at "v0.1.0"